app/grandchallenge/algorithms/models.py

```diff
@@ -1060,6 +1060,15 @@ def validate_inputs_and_execute(self, *, inputs):
                     linked_task=linked_task,
                 )
 
+    def handle_civ_error(self, interface, error_message, user=None):
+        detailed_error_message = self.detailed_error_message
+        detailed_error_message[interface.title] = error_message
```

Comment:
This construction will modify the original `self.detailed_error_message` due to how dictionaries are referenced in Python, not sure that is what you want.

Reply:
True, it doesn't matter too much since we're updating `self.detailed_error_message` right after this anyway, but it's probably still better to copy the dictionary contents into a new variable.